staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -288,7 +288,7 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			}
 
 			list, paginatedResult, err = pager.List(context.Background(), options)
-			if isExpiredError(err) {
+			if isExpiredError(err) || apierrors.IsTooLargeResourceVersion(err) {
```

Comment:
This makes the docs below, and the docs for `setIsLastSyncResourceVersionExpired` misleading, since we're setting that in cases other than for expired errors. Should we rename `setIsLastSyncResourceVersionExpired` to `setIsLastSyncResourceVersionUnavailable` or something that can make sense for both expired and toolarge errors?

Reply:
makes sense - done